make tests failing at hello_world_component

`make tests` is giving following error at commit `bee515038e`
```
libhpx_hello_world_component.so: undefined reference to `hpx::cout'
collect2: error: ld returned 1 exit status
CMakeFiles/hello_world_client.dir/build.make:127: recipe for target 'hello_world_client' failed
make[6]: * [hello_world_client] Error 1
CMakeFiles/Makefile2:104: recipe for target 'CMakeFiles/hello_world_client.dir/all' failed
make[5]: * [CMakeFiles/hello_world_client.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make[4]: * [all] Error 2
tests/unit/build/CMakeFiles/cmake_build_dir_test.make_compile.dir/build.make:57: recipe for target 'tests/unit/build/CMakeFiles/cmake_build_dir_test.make_compile' failed
make[3]: * [tests/unit/build/CMakeFiles/cmake_build_dir_test.make_compile] Error 2
CMakeFiles/Makefile2:28127: recipe for target 'tests/unit/build/CMakeFiles/cmake_build_dir_test.make_compile.dir/all' failed
make[2]: * [tests/unit/build/CMakeFiles/cmake_build_dir_test.make_compile.dir/all] Error 2
CMakeFiles/Makefile2:300: recipe for target 'CMakeFiles/tests.dir/rule' failed
make[1]: * [CMakeFiles/tests.dir/rule] Error 2
Makefile:266: recipe for target 'tests' failed
make: * [tests] Error 2
```

But `make tests` are passing at commit `69100ee061` successfully.

--- a/examples/hello_world_component/CMakeLists.txt
+++ b/examples/hello_world_component/CMakeLists.txt
@@ -22,8 +22,8 @@ if(EXISTS "${HPX_DIR}")
   add_library(hello_world_component SHARED hello_world_component.cpp)
   hpx_setup_target(
     hello_world_component
+    DEPENDENCIES iostreams_component
     TYPE COMPONENT
-    COMPONENT_DEPENDENCIES iostreams
   )
 
   add_executable(hello_world_client hello_world_client.cpp)

--- a/examples/hello_world_component/Makefile
+++ b/examples/hello_world_component/Makefile
@@ -22,6 +22,7 @@ APP_CFLAGS += -DHPX_APPLICATION_NAME=test
 
 COMP_CFLAGS = $(CXX_FLAGS)
 COMP_CFLAGS += $(shell PKG_CONFIG_PATH=$(HPX_DIR) $(PKG_CONFIG) --cflags --libs $(hpx_component))
+COMP_CFLAGS += -lhpx_iostreams$(debug_suffix)
 COMP_CFLAGS += -DHPX_COMPONENT_NAME=test_component
 
 ifneq ($(wildcard $(HPX_DIR)/*),)

